### PR TITLE
Increase minimum Python version to 3.5.3

### LINKED ISSUE #3089: Python 3.5.0-3.5.1 support
Python 3.5.0 has gone out of the conda-forge repository. 3.5.1 is still there... for now.
The anaconda repository starts directly from 3.5.4.
3.5.0 and 3.5.1 are a colossal pain in the back for typing support.
Is this a good time to increase the requirement to >= 3.5.2?
I honestly can't think how anybody could be unable to upgrade to the latest available 3.5 with minimal effort...

### PR DESCRIPTION
**Pull request by crusaderky**
Closes #3089

### CODE REVIEW

File: xarray/coding/cftime_offsets.py [hunk 1]
```diff
@@ -41,18 +41,17 @@
...
 
 import numpy as np
 
-from ..core.pycompat import TYPE_CHECKING
 from .cftimeindex import CFTimeIndex, _parse_iso8601_with_reso
 from .times import format_cftime_datetime
 
 if TYPE_CHECKING:
-    from typing import ClassVar, Optional
+    from typing import ClassVar  # Requires Python >= 3.5.3
```
Comment (shoyer) [inline xarray/coding/cftime_offsets.py]: We can bump all the way to Python 3.5.3 if you prefer :)

Comment (crusaderky) [reply xarray/coding/cftime_offsets.py]: Dude, if it were for me I'd bump it to 3.7 😂
It's up for you to decide...

Comment (shoyer) [reply xarray/coding/cftime_offsets.py]: Let's bump it up to the lowest 3.5.x version that has all the typing utilities we want.

Comment (crusaderky) [reply xarray/coding/cftime_offsets.py]: Bumped to 3.5.3 and ready to merge

**Comment by rabernat** [comment]
As far as I can tell, @shoyer's review has been addressed. Sprinters at scipy would like to have this, so I'm going to merge.

**Pull request merged**